### PR TITLE
SOLR-16781: Disable <lib/> by default on 9.x

### PR DESCRIPTION
**Pull request by gerlowskija**
https://issues.apache.org/jira/browse/SOLR-16781

# Description

`<lib/>` usage will now log a warning by default for future 9.x releases.  Wary users can re-enabled the feature by specifying a sysprop: `solr.config.lib.enabled=true`.

This PR is a companion to https://github.com/apache/solr/pull/2875, which makes more aggressive changes (removing `<lib>` altogether) on the 'main' branch.  As such, this PR is only intended for branch_9x, and #2875 is only intended for 'main'.

# Tests

New test in `TestConfigWithLibDisabled`

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://github.com/apache/solr/blob/main/CONTRIBUTING.md) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended, not available for branches on forks living under an organisation)
- [x] I have run `./gradlew check`.
- [x] I have added tests for my changes.
- [x] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

File: solr/core/src/java/org/apache/solr/cli/RunExampleTool.java [hunk 1]
```diff
@@ -620,10 +620,15 @@ protected Map<String, Object> startSolr(
     if (!isWindows && cwdPath.length() > 1 && solrHome.startsWith(cwdPath))
       solrHome = solrHome.substring(cwdPath.length() + 1);
 
+    final var syspropArg =
```
Comment (HoustonPutman) [inline solr/core/src/java/org/apache/solr/cli/RunExampleTool.java:623]: If you are adding this, should the lib directive be removed from the example configSet?

File: solr/core/src/test/org/apache/solr/cli/TestSolrCLIRunExample.java [hunk 2]
```diff
@@ -70,6 +72,7 @@ public static void beforeClass() {
         Paths.get(".").toAbsolutePath().toString().contains(" "));
     // to be true
     System.setProperty("solr.directoryFactory", "solr.NRTCachingDirectoryFactory");
+    System.setProperty(LIB_ENABLED_SYSPROP, "true");
```
Comment (HoustonPutman) [inline solr/core/src/test/org/apache/solr/cli/TestSolrCLIRunExample.java:75]: If we remove the lib directives in the example ConfigSet, these should be able to be removed, right?

File: solr/core/src/java/org/apache/solr/cli/RunExampleTool.java (hunk 1 above)
Comment (gerlowskija) [reply solr/core/src/java/org/apache/solr/cli/RunExampleTool.java:623]: It's already gone, afaict.  @epugh removed it (unrelated to this change) in [a recent PR here](https://github.com/apache/solr/pull/2861/files#diff-1b6e36f0bb5a2f3801d561bd2fa32cb1e354e8342f6cf80e6e6dbfa9ed54d83c)

File: solr/core/src/test/org/apache/solr/cli/TestSolrCLIRunExample.java (hunk 2 above)
Comment (gerlowskija) [reply solr/core/src/test/org/apache/solr/cli/TestSolrCLIRunExample.java]: Good catch, fixed!

**Pull request merged**